src/ngraph/op/allreduce.hpp

```diff
@@ -28,13 +28,26 @@ namespace ngraph
         public:
             NGRAPH_API
             static const std::string type_name;
+            const Reduce_t reduce_type;
             const std::string& description() const override { return type_name; }
             AllReduce();
-            AllReduce(const std::shared_ptr<Node>& arg);
+            AllReduce(const std::shared_ptr<Node>& arg, const Reduce_t reduce_type = Reduce_t::sum);
 
             void validate_and_infer_types() override;
 
             std::shared_ptr<Node> copy_with_new_args(const NodeVector& new_args) const override;
+            std::string getReduceTypeString() const {
```

Comment:
nGraph house style would be `get_reduce_type_string`. In any case, I think it's better if this is a free function on `Reduce_t`, rather than a member of `AllReduce`. Or, better, define an `operator<<` overload for `Reduce_t`.